File: interface/src/main/java/cn/itcast/core/service/SeckillGoodsService.java
package cn.itcast.core.service;

import cn.itcast.core.pojo.entity.PageResult;
import cn.itcast.core.pojo.good.Goods;
import cn.itcast.core.pojo.item.Item;
import cn.itcast.core.pojo.seckill.SeckillGoods;
import cn.itcast.core.pojo.seller.Seller;

import java.math.BigDecimal;
import java.util.List;

/**
 * 商品秒杀申请
 */
public interface SeckillGoodsService {
    //1. 获取商家名称
    Seller findName(String userName);

    //2. 根据商家用户名搜索商家拥有的商品
    List<Goods> findByParentId(String name);

    //3. 根据商品id查询库存集合
    List<Item> findByParent(Long id);

    //4. 根据库存集合id查询价格
    BigDecimal findPriceOld(Long id);

    //5. 添加秒杀申请
    void add(SeckillGoods seckillGoods);

    //6.运营商查询所有未审核商品
    PageResult search(Integer page, Integer rows, SeckillGoods title);

    //7. 查询所有商户名字
    List<Seller> findAllName();

    //8. 查询所有商品名字
    List<Goods> findAllGoods();

    //9.查询所有库存集合名字
    List<Item> finditemname();

    //10. 根据秒杀商品id改变数据库中秒杀商品状态
    void updateStatus(Long[] ids, String status);
}

File: web_shop/src/main/webapp/js/service/seckill_goodsService.js
//服务层
app.service('seckill_goodsService', function ($http) {


    //查询商家名称
    this.findName = function () {
        return $http.post('../seckillgoods/findName.do');
    }

    //查询商品
    this.findByParentId = function () {
        return $http.post('../seckillgoods/findByParentId.do');
    }

    //查询库存集合
    this.findByParent = function (id) {
        return $http.post('../seckillgoods/findByParent.do?id=' + id);
    }

    //查询价格
    this.findPriceOld = function (id) {
        return $http.post('../seckillgoods/findPriceOld.do?id=' + id);
    }

    //增加
    this.add = function (seckillMap) {
        return $http.post('../seckillgoods/add.do', seckillMap);
    }
    //修改
    this.update = function (entity) {
        return $http.post('../seckillgoods/update.do', entity);
    }
});
